bitcoind-regtest/src/Bitcoin/Core/Regtest/Generator.hs

```diff
@@ -52,129 +89,240 @@ generateWithTransactions ::
+            <$> RPC.listUnspent
                 Nothing
                 Nothing
                 Nothing
                 Nothing
-        replicateM 100 generatorGenerate
-
-    generatorGenerate = do
-        coinbaseOutput <- lift $ do
-            addr <- newAddress
-            RPC.generateToAddress 1 addr Nothing
-                >>= fmap (processCoinbase . head . blockTxns)
-                    . RPC.getBlock
-                    . head
-        modify $ \state ->
-            state
-                { generatorUnspentCoinbases =
-                    generatorUnspentCoinbases state |> coinbaseOutput
-                }
+                (ListUnspentOptions Nothing Nothing Nothing Nothing)
+    extractPointAmount =
+        (OutPoint <$> RPC.outputTxId <*> RPC.outputVOut) &&& RPC.outputAmount
 
-    newAddress = RPC.getNewAddress Nothing Nothing
+getBalance :: BitcoindClient Word64
+getBalance = RPC.balanceDetailsTrusted . RPC.balancesMine <$> RPC.getBalances
+
+getSeriesCount :: Monad m => StateT [TxSeries] m Int
+getSeriesCount = gets length
+
+type TxSeries = [Tx]
 
-    sendTransactions = do
-        lift $ RPC.walletPassphrase password (2 * blockInterval)
-        feeRate <- getMeanFeeRate <$> lift RPC.getBlockCount
-        splitOutputTxs <- spendSplitOutputs feeRate
-        coinbaseSplitTx <- popCoinbase >>= traverse (splitCoinbaseOutput feeRate)
-        lift . mapM_ (`RPC.sendTransaction` Nothing) $ splitOutputTxs
-        lift . mapM_ (`RPC.sendTransaction` Nothing) $ coinbaseSplitTx
-
-    popCoinbase = do
-        state <- get
-        case generatorUnspentCoinbases state of
-            next :<| rest -> Just next <$ put state{generatorUnspentCoinbases = rest}
-            _ -> pure Nothing
-
-    appendSpendable tx = modify $ \state ->
-        state
-            { generatorSpendableOutputs =
-                generatorSpendableOutputs state
-                    <> zipWith
-                        (mkSpendableOutput (txHash tx))
-                        [0 ..]
-                        (outValue <$> txOut tx)
+-- | Create the funding anchor for a tx series and add the tx series to the internal state
+fund ::
+    -- | Output count per transaction
+    Int ->
+    -- | Fee rate
+    Word64 ->
+    -- | Sats to allocate
+    Word64 ->
+    StateT [TxSeries] BitcoindClient TxHash
+fund nOutputs feeRate sats = do
+    (value, outPoint) <- lift createFundingOutput
+    modify' $ (:) (take seriesLength $ txSeries nOutputs feePerOutput value outPoint)
+    pure $ H.outPointHash outPoint
+  where
+    createFundingOutput = do
+        tx <- createRoot >>= (`RPC.getRawTransaction` Nothing)
+        (ix, txOut) <-
+            maybe notFound pure
+                . find ((== easySpendScriptOutput) . H.scriptOutput . snd)
+                . zip [0 ..]
+                $ H.txOut tx
+        pure
+            ( H.outValue txOut
+            , OutPoint (H.txHash tx) ix
+            )
+    notFound = error "Funding output not found!"
```

Comment:
Should this be an exception? Or is this truly impossible to happen?

Reply:
Ah!  I think we can simplify this code a bit now that we are completely spending exactly one output in this step.